src/Symfony/Component/Webhook/Server/GeneratorInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Webhook\Server;
+
+use Symfony\Component\HttpClient\HttpOptions;
+use Symfony\Component\RemoteEvent\RemoteEvent;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 6.3
+ */
+interface GeneratorInterface
+{
+    public function generate(RemoteEvent $event, string $secret, HttpOptions $options): void;
+}
```

Comment:
The scope of this interface is not obvious, both its name and its API feel too broad. I'd either refine it or remove it completely

Reply:
I had the same feeling when reading the code. At least what the implementing classes do is not really generating anything but enriching/configuring/bootstrapping an HTTP request

Reply:
Renamed to `RequestConfiguratorInterface`